pkg/controllers/management/auth/psautils/common.go

```diff
@@ -0,0 +1,28 @@
+package psautils
+
+import (
+	"github.com/rancher/rancher/pkg/apis/management.cattle.io"
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	rbacmgmt "k8s.io/kubernetes/plugin/pkg/auth/authorizer/rbac"
+)
+
+const (
+	UpdatepsaVerb = "updatepsa"
+)
+
+// IsPSAAllowed get a list of roletemplates to verify if updatepsa verb is enabled on their rules
+func IsPSAAllowed(roleTemplates []*v3.RoleTemplate) bool {
+	auth := authorizer.AttributesRecord{
+		Verb:            UpdatepsaVerb,
+		APIGroup:        management.GroupName,
+		Resource:        v3.ProjectResourceName,
+		ResourceRequest: true,
+	}
+	for _, rt := range roleTemplates {
+		if rbacmgmt.RulesAllow(auth, rt.Rules...) {
```

Comment:
what if a RoleTemplate has inherited RoleTemplatee? Should we check inherited rules too?

Reply:
right, on the other [PR](https://github.com/rancher/rancher/pull/49777/files#diff-362676b310f18ae93af844d6ea991a68489832b13c7fc7b69835f448adfaca1dR1011) I was using `gatherAndDedupeRoles` function which is not provided in this package.
we should find a way to make it available even here (might be moving it to a dedicated package or reimplementing by reimplementing this).